apps/wallet/src/ui/app/hooks/useAccountByAddress.ts

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useMemo } from 'react';
+import { useAccounts } from './useAccounts';
+
+export function useAccountByAddress(accountAddress?: string | null) {
+	const allAccountsData = useAccounts();
+	const account = useMemo(
+		() =>
+			(accountAddress && allAccountsData.data?.find(({ address }) => address === accountAddress)) ||
+			null,
+		[allAccountsData.data, accountAddress],
+	);
+	return { ...allAccountsData, data: account };
```

Comment:
do we need to include all accounts data here?

Reply:
maybe to use isLoading etc. might revisit it